packages/shared-metrics/src/util/nativeModuleRetry.js

```diff
@@ -171,51 +170,68 @@ function copyPrecompiled(opts, loaderEmitter, callback) {
+                logger.warn(`Copying the precompiled build for ${opts.nativeModuleName} ${label} failed.`, cpErr);
+                callback(false);
+                return;
+              }
 
-            // We have unpacked and copied the correct precompiled native addon. The next attempt to require the
-            // dependency should work.
-            //
-            // However, we must not use any of the paths from which Node.js has tried to load the module before (that
-            // is, node_modules/${opts.nativeModuleName}). Node.js' module loading infrastructure
-            // (lib/internal/modules/cjs/loader.js and lib/internal/modules/package_json_reader.js) have built-in
-            // caching on multiple levels (for example, package.json locations and package.json contents). If Node.js
-            // has tried unsuccessfully to load a module or read a package.json from a particular path, it will remember
-            // and not try to load anything from that path again (a `false` will be put into the cache for that cache
-            // key). Instead, we force a new path, by adding precompiled to the module path and use the absolute path to
-            // the module to load it.
-            opts.loadFrom = targetDir;
-            callback(true);
-          }
-        );
-      })
-      .catch(tarErr => {
-        logger.warn(`Unpacking the precompiled build for ${opts.nativeModuleName} ${label} failed.`, tarErr);
-        callback(false);
-      });
+              // We have unpacked and copied the correct precompiled native addon. The next attempt to require the
+              // dependency should work.
+              //
+              // However, we must not use any of the paths from which Node.js has tried to load the module before (that
+              // is, node_modules/${opts.nativeModuleName}). Node.js' module loading infrastructure
+              // (lib/internal/modules/cjs/loader.js and lib/internal/modules/package_json_reader.js) have built-in
+              // caching on multiple levels (for example, package.json locations and package.json contents). If Node.js
+              // has tried unsuccessfully to load a module or read a package.json from a particular path,
+              // it will remember and not try to load anything from that path again (a `false` will be
+              // put into the cache for that cache key). Instead, we force a new path, by adding precompiled
+              // to the module path and use the absolute path to the module to load it.
+              opts.loadFrom = targetDir;
+              callback(true);
+            }
+          );
+        })
+        .catch(tarErr => {
+          logger.warn(`Unpacking the precompiled build for ${opts.nativeModuleName} ${label} failed.`, tarErr);
+          callback(false);
+        });
+    } catch (err) {
+      logger.warn(`Tar extraction for precompilation has failed for ${opts.nativeModuleName}.`, err);
+      callback(false);
```

Comment:
*question:* What makes this new try-catch block around the whole extraction necessary? Is this for the case when requiring node-tar@6 on an older Node.js version actually works but calling functions from it does not? In that case, it might be an argument to simply not requiring in Node.js < 10 in the first place.